src/dft/backends/cufft/commit.cpp

```diff
@@ -44,15 +44,20 @@ namespace detail {
 template <dft::precision prec, dft::domain dom>
 class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
 private:
+    using scalar_type = typename dft::detail::commit_impl<prec, dom>::scalar_type;
+
     // For real to complex transforms, the "type" arg also encodes the direction (e.g. CUFFT_R2C vs CUFFT_C2R) in the plan so we must have one for each direction.
     // We also need this because oneMKL uses a directionless "FWD_DISTANCE" and "BWD_DISTANCE" while cuFFT uses a directional "idist" and "odist".
     // plans[0] is forward, plans[1] is backward
     std::array<std::optional<cufftHandle>, 2> plans = { std::nullopt, std::nullopt };
     std::array<std::int64_t, 2> offsets;
+    // For using a buffer external workspace, the buffer object itself needs to be kept between compute calls such that accessors can be used.
+    std::optional<sycl::buffer<scalar_type, 1>> buffer_workspace;
```

Comment:
Is this needed? I don't think it's being used since it is already part of `commit_impl` and `external_workspace_helper`

Reply:
Its leftover from where my initial attempts didn't neatly fit into the existing code. Removed in [1da13ee](https://github.com/oneapi-src/oneMKL/pull/411/commits/1da13eef0fbb663fcadc1abbcc15ad930f977589).